### ISSUE TITLE
Metadata cannot preserve using Exiftool if the folder path contain Chinese characters

### ISSUE THREAD

**Issue by barrykwok** [JacobDev1/xl-converter#47]
### Version

1.0.1

### OS

Windows

### Description

if the image folder path contain Chinese characters, then converted images won't have any metadata copied from the source image

### Steps to Reproduce

1. create a folder containing Chinese character and copy an image file with metadata into it, for example, C:\測試影像\
2. convert the image and preserve metadata using Exiftool
3. open the converted image, all metadata is blank
4. create another folder which only contain English characters and copy the source image file into it, for example, C:\TestingImages\
5. convert the image and preserve metadata using Exiftool
6. open the converted image from the folder created in step 4, all metadata is preserved

### Expected Result

all metadata can preserve in converted image

### Current Result

none of the metadata is preserved in converted image

### Input Formats

TIF

### Output Format

JPEG

**Comment by JacobDev1** [comment]
This is a bug in the ExifTool itself. While ExifTool does support UTF-8 on Windows, Asian characters don't seem to work.

When you try to read an image with the name you suggested, it returns:

```txt
No matching files
```

Embedding a manifest breaks the executable, so I can't patch it.

```txt
Can't locate PAR.pm in @INC (you may need to install the PAR module) (@INC contains: .) at -e line 629.
```

There is nothing I can do about this. If you need this fixed, just contact the ExifTool developer.

Getting back to my application: Unfortunately, `cjpegli` encoder does not have any metadata options, so `Encoder - Preserve` will not work.

**XnConvert** has JPEGLI and can handle metadata well, so use that instead.

**Comment by barrykwok** [comment]
You can use an argument file to pass the arguments to ExifTool, instead of passing them direct in command line, it can handle UTF-8 file path with this method
https://exiftool.org/exiftool_pod.html#ARGFILE

**Comment by JacobDev1** [comment]
I prototyped a workaround based on argfiles. It seems to work at first glance, and may appear in the next version.

Unfortunately, there is a caveat. If you install this program in a path with Unicode specific characters (on Windows), ExifTool will stop working. That's because ExifTool fails to run if you call it using an absolute path with Unicode characters.

It probably fails to parse argv[0].

**Comment by JacobDev1** [comment]
A fix for it will appear in version `1.0.2`. This version will feature an improved ExifTool handling overall.

The only limitation is the path to the program cannot contain special characters.

Below is an error message that ExifTool displays when run from a path with special characters. This error is difficult to reproduce, as most of the time ExifTool returns an empty string instead.

```
assertion failed: open my_prog
assertion failed: MZ magic bytes
assertion failed: PE header
assertion failed: IMAGE_NT_OPTIONAL_HDR_MAGIC
```

XL Converter will verify that ExifTool is working correctly and notify the user if it isn't.
